### DISCUSSION TITLE
Submit Button - only show when field completed

### DISCUSSION THREAD

**Discussion by blanch80** [liedekef/events-made-easy#166] [Q&A]
Hi all,

I have a booking form that shows dynamic data depending on what is answered in custom field.

For example - are you in the UK or Ireland. Depending on the choice made a different 'next' question appears - all working as expected. The fields are all mandatory and this is where I have a slight issue.

Because of the very slight delay of the template loading based on the dynamic data condition, it is possible for the user to submit the form before those extra fields have loaded, even though they're mandatory.

If they wait until it loads and try to submit the booking, the mandatory field check works. If they click submit before the template based on the dynamic data appears, the booking goes through.

Any ideas on a way round this? Is there a way to hide the submit button until the selection has been made in the template that loads based on the dynamic data condition?

Thanks,
David.

**Comment by liedekef** [comment]
Normally the submit button is hidden, but maybe some older change created an issue here. I'll check this.

**Comment by liedekef** [comment]
Can you check if this version of eme.js works? https://github.com/liedekef/events-made-easy/commit/c65cd81a37f958ca7a1848a8316b72b2a6da7528

**Reply by liedekef** [reply]
This version should be better:
https://github.com/liedekef/events-made-easy/commit/2763c7f35150fea708dc77ba6003f956c929cc02

**Reply by blanch80** [reply]
I will do as soon as I get the chance, thanks Franky.

**Comment by blanch80** [comment]
Hi Franky,

Doesn't seem to have worked. I've tried it with #_submit in the template and not in the template.

The main template is...

`[eme_if tag='#_USER_HAS_ROLE{#_ATT{role}}' value='1']#_SEATS
<table class='eme-rsvp-form'>
    <tr><th scope='row'>Do you work for a Practice or PCN?*:</th><td>#_FIELD{Condition-Practice-PCN}</td></tr>
</table>
<table class='eme-rsvp-form'>
    <tr><th scope='row'>First Name*:</th><td>#REQ_FIRSTNAME</td></tr>#_SEATS 
    <tr><th scope='row'>Last Name*:</th><td>#_LASTNAME</td></tr>
    <tr><th scope='row'>E-Mail*:</th><td>#_EMAIL</td></tr>
</table>
#_SUBMIT
[/eme_if]
[eme_if2 tag='#_USER_HAS_ROLE{#_ATT{role}}' value='0']<strong>You are unable to book a place on this event. If you would like more details, please contact <a href="mailto:necsu.csstbookings@nhs.net">necsu.csstbookings@nhs.net</a> with your own, and your organisation details.</strong>[/eme_if2]`

And loads this template if 'PCN' is chosen from #_FIELD{Condition-Practice-PCN}

`[eme_if tag='#_USER_HAS_ROLE{nenc_county_durham}' value='1']
<table class="eme-rsvp-form">
<tbody>
<tr>
<th scope="row">Durham PCN Name*:</th>
<td>#_FIELD{PCN-List-Durham}</td>
</tr>
</tbody>
</table>
[/eme_if]
[eme_if tag='#_USER_HAS_ROLE{nenc_northumberland}' value='1']
<table class="eme-rsvp-form">
<tbody>
<tr>
<th scope="row">Northumberland PCN Name*:</th>
<td>#_FIELD{PCN-List-Northumberland }</td>
</tr>
</tbody>
</table>
[/eme_if]
[eme_if tag='#_USER_HAS_ROLE{nenc_north_tyneside}' value='1']
<table class="eme-rsvp-form">
<tbody>
<tr>
<th scope="row">North Tyneside PCN Name*:</th>
<td>#_FIELD{PCN-List-North-Tyneside}</td>
</tr>
</tbody>
</table>
[/eme_if]
[eme_if tag='#_USER_HAS_ROLE{nenc_newcastle_gateshead}' value='1']
<table class="eme-rsvp-form">
<tbody>
<tr>
<th scope="row">Newcastle-Gateshead PCN Name*:</th>
<td>#_FIELD{PCN-List-Newcastle-Gateshead}</td>
</tr>
</tbody>
</table>
[/eme_if]
[eme_if tag='#_USER_HAS_ROLE{nenc_south_tyneside}' value='1']
<table class="eme-rsvp-form">
<tbody>
<tr>
<th scope="row">South Tyneside PCN Name*:</th>
<td>#_FIELD{PCN-List-South-Tyneside}</td>
</tr>
</tbody>
</table>
[/eme_if]
[eme_if tag='#_USER_HAS_ROLE{nenc_sunderland}' value='1']
<table class="eme-rsvp-form">
<tbody>
<tr>
<th scope="row">Sunderland PCN Name*:</th>
<td>#_FIELD{PCN-List-Sunderland}</td>
</tr>
</tbody>
</table>
[/eme_if]
[eme_if tag='#_USER_HAS_ROLE{nenc_tees_valley}' value='1']
<table class="eme-rsvp-form">
<tbody>
<tr>
<th scope="row">Tees Valley PCN Name*:</th>
<td>#_FIELD{PCN-List-Tees-Valley}</td>
</tr>
</tbody>
</table>
[/eme_if]
[eme_if tag='#_USER_HAS_ROLE{hny_york_city}' value='1']
<table class="eme-rsvp-form">
<tbody>
<tr>
<th scope="row">York City PCN Name*:</th>
<td>#_FIELD{PCN-List-York-City}</td>
</tr>
</tbody>
</table>
[/eme_if]
[eme_if tag='#_USER_HAS_ROLE{hny_north_yorkshire}' value='1']
<table class="eme-rsvp-form">
<tbody>
<tr>
<th scope="row">North Yorkshire PCN Name*:</th>
<td>#_FIELD{PCN-List-North-Yorkshire}</td>
</tr>
</tbody>
</table>
[/eme_if]
[eme_if tag='#_USER_HAS_ROLE{dby_amber_valley,dby_bolsover_and_ne_derbyshire,dby_chesterfield,dby_derby_city,dby_derbyshire_dales,dby_derbyshire_dales,dby_high_peak,dby_south_derbyshire,dby_south_derbyshire}' value='1']
<table class="eme-rsvp-form">
<tbody>
<tr>
<th scope="row">Derbyshire PCN Name*:</th>
<td>#_FIELD{PCN-List-Derbyshire}</td>
</tr>
</tbody>
</table>
[/eme_if]`

If #_SUBMIT isn't included in the template it still appears, just a bit lower down on the page.

Hope you can help.

![Conditions](https://user-images.githubusercontent.com/123082349/236242090-ddaf7cf3-e319-4a3a-9d23-23bb4c81e046.png)
![Form with submit in template](https://user-images.githubusercontent.com/123082349/236242078-786d49ed-a0a6-48a1-bbda-f72f6afeff89.png)
![Form without submit in template code](https://user-images.githubusercontent.com/123082349/236242086-00ad11eb-5cad-4050-864b-e382f00db549.png)

**Reply by liedekef** [reply]
Make sure the older Js is not cached. My tests for members work fine

**Reply by liedekef** [reply]
Btw: I tested some more and simplified the code more. Also tested with rsvp + dynamic fields, now seem to work as expected. Please try the latest eme.js from the main code.

**Comment by blanch80** [comment]
Sorry for the delay in getting back to you Franky - was on annual leave. Tried the new js file and it works a treat, thank you.
